### PR TITLE
Add docs on symbolic input

### LINKED ISSUE #747: Improve documentation on how to provide symbolic input to Manticore
Being able to provide symbolic input to a program is a core feature of Manticore, however there is no direct documentation on this, which leads to poor user experiences.

We casually mention it in the readme: https://github.com/trailofbits/manticore#cli but there is no documentation that directly states: to give symbolic argv, do this, to give symbolic stdin, do this, to give symbolic file input, do this.

### PR DESCRIPTION
**Pull request by mossberg**
Fixes #747 
looks like this:

image

---
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/750)

### CODE REVIEW

**Comment by mossberg** [comment]
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[docs/syminput.rst, line 11 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/750#-L5GSGQGluU7ITH4vSnc:-L5G_bfxkqbY-LKS8PT8:b-896fix) ([raw file](https://github.com/trailofbits/manticore/blob/8233fb04fb9358079db7e24f8c07a4433915ec59/docs/syminput.rst#L11)):*
<details><summary><i>Previously, yan (Yan Ivnitskiy) wrote…</i></summary><blockquote>

I'm not sure this is clear, it sounds like it can be used to create constrained bytes. Maybe reword as "This allows the user to make input that mixes concrete and symbolic bytes"
</blockquote></details>

Done.

---

*[docs/syminput.rst, line 19 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/750#-L5GSWUclRzw1izTRZJB:-L5G_ZGNQC6ZbTkIsdE5:b-896fix) ([raw file](https://github.com/trailofbits/manticore/blob/8233fb04fb9358079db7e24f8c07a4433915ec59/docs/syminput.rst#L19)):*
<details><summary><i>Previously, yan (Yan Ivnitskiy) wrote…</i></summary><blockquote>

"on the ~~Manticore~~ command line"?
</blockquote></details>

Done.

---

*[docs/syminput.rst, line 32 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/750#-L5GSfkmGa5OlChS55qu:-L5G_cHxEoBxKf8cVWy-:b-896fix) ([raw file](https://github.com/trailofbits/manticore/blob/8233fb04fb9358079db7e24f8c07a4433915ec59/docs/syminput.rst#L32)):*
<details><summary><i>Previously, yan (Yan Ivnitskiy) wrote…</i></summary><blockquote>

".. 256 **bytes** of.."
</blockquote></details>

Done.

---

*[docs/syminput.rst, line 41 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/750#-L5GSp9ofCEZDDek5Leq:-L5GZZEvmk4LYlisUEg2:bl8f5ju) ([raw file](https://github.com/trailofbits/manticore/blob/8233fb04fb9358079db7e24f8c07a4433915ec59/docs/syminput.rst#L41)):*
<details><summary><i>Previously, yan (Yan Ivnitskiy) wrote…</i></summary><blockquote>

"..create the files that will be opened by the analyzed program **, add them via the `--file` flag, and and fill** them with.." (and potentially remove the ref to `--file` below. This way we have a summary of how to add it up front in the paragraph.
</blockquote></details>

the `--file` flag only applies to command line use though, whereas currently the first paragraph applies to both cli and api.

will fix the "filling" typo.

---

*[docs/syminput.rst, line 47 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/750#-L5GTI0FVSaa0mriSWC5:-L5G_dg_yuZ4ybqGOdJn:b-896fix) ([raw file](https://github.com/trailofbits/manticore/blob/8233fb04fb9358079db7e24f8c07a4433915ec59/docs/syminput.rst#L47)):*
<details><summary><i>Previously, yan (Yan Ivnitskiy) wrote…</i></summary><blockquote>

Use `:meth:` ref for `add_symbolic_file`?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/trailofbits/manticore/750)*

**Comment by yan** [comment]
:lgtm:

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/trailofbits/manticore/750#-:-L5GaMeqJT8woDtwrv_d:bnfp4nl)*

**Comment by yan** [comment]
Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/trailofbits/manticore/750)*
